Managers/UInputMan.cpp

```diff
@@ -837,8 +839,19 @@ namespace RTE {
 			return;
 		}
 
-		if (g_InActivity) {
-			if (AnyStartPress(false)) {
+		if (g_InActivity) {	
+			bool BuyMenuIsOpen = 0;
+			GameActivity* temp = dynamic_cast<GameActivity*>(g_ActivityMan.GetActivity());
+
+			// Make ESC key open BuyMenuGUIs if any are open rather than pausing the game.
+			for (short player = Players::PlayerOne; player < Players::MaxPlayerCount; player++) {
+				if (temp->GetBuyGUI(player)->IsVisible()) {
+					BuyMenuIsOpen = true;
+					break;
+				}
+			}
+
+			if (!BuyMenuIsOpen && AnyStartPress(false)) {
```

Comment:
This method is called every time UInputMan's Update method is run, which means you've got this stuff running each frame. It's not a huge deal, cause it's all relatively small stuff, but I see no reason not to make it a bit smarter.
Rather than running all your checks and then doing `if (AnyStartPress(false))` you could do
```
if (AnyStartPress(false)) {
	//Code here to figure out if buymenu is open
	if (!BuyMenuIsOpen) {
		g_ActivityMan.PauseActivity();
		return
	}
}
```